### PR TITLE
Benefits - Sprint 2

### PR DESCRIPTION
**Pull request by MilanPodunavac**
Controller in Integration is BenefitController, contains endpoints for all repo methods
RabbitMQ logic is in MicroService BenefitRabbitMQService
Normal repo with a couple of new methods for filtering

In pharmacy, everything simplyfied, BenefitController and repo

### CODE REVIEW

File: Integration/IntegrationApi/Controllers/SftpController.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                _medicineConsumptionMasterService.CreateConsumptionReportInTimeRange(dto.TimeRange);
+            try
+            {
+                _sftpMasterService.SaveMedicineReportToSftpServer(report);
+            }
+            catch (Exception e)
+            {
+                return BadRequest("Failed to send contact sftp server");
+            }
+            /*RestClient client = new RestClient();
```
Comment (AleksicNikola00) [inline Integration/IntegrationApi/Controllers/SftpController.cs]: Delete comments

File: Integration/IntegrationClassLib/MasterServices/SftpMasterService.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                sftpClient.Disconnect();
+                fileStream.Close();
+            }
+            catch (Exception e)
+            {
+                throw e;
+            }
+        }
+
+        /*public bool TestSftpDownload()
```
Comment (AleksicNikola00) [inline Integration/IntegrationClassLib/MasterServices/SftpMasterService.cs]: Also delete comments.

File: Integration/IntegrationClassLib/MasterServices/SftpMasterService.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            try
+            {
+                StreamWriter fileSaveStream = new StreamWriter(path);
+                string jsonString = JsonConvert.SerializeObject(consumptionReport);
+                fileSaveStream.Write(jsonString);
+                fileSaveStream.Close();
+            }
+            catch (Exception e)
+            {
+                throw e;
```
Comment (AleksicNikola00) [inline Integration/IntegrationClassLib/MasterServices/SftpMasterService.cs]: Consider not throwing exceptions in catch block.

File: Pharmacy/PharamcyApi/Controllers/BenefitController.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                    };
+                    var body = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(message));
+
+                    channel.BasicPublish(exchange: "benefits",
+                        routingKey: "",
+                        basicProperties: null,
+                        body: body);
+                }
+            }
+            catch (Exception e)
```
Comment (AleksicNikola00) [inline Pharmacy/PharamcyApi/Controllers/BenefitController.cs]: Same thing here with exceptions.

**Review by AleksicNikola00 — DISMISSED** [review]

**Review by AleksandarRadisic — APPROVED** [review]

**Review by AleksicNikola00 — APPROVED** [review]

**Pull request merged**